jsonrpc/event_listener_test.go

```diff
@@ -37,3 +54,11 @@ func (l *CountingEventListener) OnRequestFailed(method string, data any) {
 		data:   data,
 	})
 }
+
+func (l *CountingEventListener) OnNewConnection() {
+	l.OnConnectionCalls++
+}
+
+func (l *CountingEventListener) OnDisconnect() {
+	l.OnConnectionCalls--
```

Comment:
I'm not sure that we can do ++ or -- here because these methods called concurrently. Should it be atomic types instead ?